client/__tests__/util/dataframe/dataframe.test.js

```diff
@@ -293,6 +316,222 @@ describe("dataframe factories", () => {
       expect(dfB.icol(i).asArray()).toEqual(dfA.icol(i).asArray());
     }
   });
+
+  describe("withCol", () => {
+    test("KeyIndex", () => {
+      const df = new Dataframe.Dataframe(
+        [2, 2],
+        [["red", "blue"], [true, false]],
+        null,
+        new Dataframe.KeyIndex(["colors", "bools"])
+      );
+      const dfA = df.withCol("numbers", [1, 0]);
+
+      expect(dfA).toBeDefined();
```

Comment:
I'll be on the lookout for a better way to run tests when many of the lines within the test case are almost equivalent. These are getting hard to read.

Reply:
I am open to suggestions....

Reply:
I'll keep my eyes open. I don't have any ideas currently.

Reply:
I try to use before/after when it makes sense, but in this case it just doesn't...

Reply:
It's the number of similar expect statements across tests that is overwhelming to my brain. It is fine technically, I just wish there was a better way.